PROTO_tests/tests/segarray_test.py

```diff
@@ -0,0 +1,694 @@
+        with pytest.raises(TypeError):
+            sa.append(err_sa)
+
+        edge_seg, edge_val = self.make_segarray_edge(dtype)
+        edge_sa = ak.SegArray(ak.array(edge_seg), ak.array(edge_val))
+
+        result = sa.append(edge_sa)
+
+        assert isinstance(result, ak.SegArray)
+        assert result.size == (sa.size + edge_sa.size)
+        assert result.lengths.to_list() == (sa.lengths.to_list() + edge_sa.lengths.to_list())
+        assert result.segments.to_list() == np.concatenate([seg_np, edge_seg + val_np.size]).tolist()
+        assert result.values.to_list() == sa.values.to_list() + edge_sa.values.to_list()
+        assert result.to_list() == (sa.to_list() + edge_sa.to_list())
+
+        result = edge_sa.append(sa)
+        assert isinstance(result, ak.SegArray)
+        assert result.size == (edge_sa.size + sa.size)
+        assert result.lengths.to_list() == (edge_sa.lengths.to_list() + sa.lengths.to_list())
+        assert result.segments.to_list() == np.concatenate([edge_seg, seg_np + edge_val.size]).tolist()
+        assert result.values.to_list() == edge_sa.values.to_list() + sa.values.to_list()
+        assert result.to_list() == (edge_sa.to_list() + sa.to_list())
+
+        # test axis=1
+        if dtype != ak.str_:  # TODO - updated to run on strings once #2646 is complete
+            seg_np, val_np = self.make_segarray(size, dtype)
+            sa2 = ak.SegArray(ak.array(seg_np), ak.array(val_np))
+
+            result = sa.append(sa2, axis=1)
+            assert isinstance(result, ak.SegArray)
+            assert result.size == sa.size
+            assert result.lengths.to_list() == [
+                x + y for (x, y) in zip(sa.lengths.to_list(), sa2.lengths.to_list())
+            ]
+            assert result.to_list() == [x + y for (x, y) in zip(sa.to_list(), sa2.to_list())]
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NO_STR)
+    def test_append_single(self, size, dtype):
+        # TODO - add testing for empty segments with issue #2650
+        seg_np, val_np = self.make_segarray(size, dtype)
+        sa = ak.SegArray(ak.array(seg_np), ak.array(val_np))
+        to_append = self.build_single_append_array(sa.size, dtype)
+        result = sa.append_single(ak.array(to_append))
+
+        assert isinstance(result, ak.SegArray)
+        assert result.size == sa.size
+        assert result.lengths.to_list() == (sa.lengths + 1).to_list()
+        sa_list = sa.to_list()
+        for i, s in zip(range(len(sa_list)), sa_list):
+            s.append(to_append[i])
+        assert result.to_list() == sa_list
+
+        # test single value
+        to_append = self.get_append_scalar(dtype)
+        result = sa.append_single(to_append)
+        assert result.size == sa.size
+        assert result.lengths.to_list() == (sa.lengths + 1).to_list()
+        sa_list = sa.to_list()
+        for i, s in zip(range(len(sa_list)), sa_list):
+            s.append(to_append)
```

Comment:
`i` is never used here

Reply:
Corrected this case. `i` is not necessary.